rpc/block.go

```diff
@@ -217,6 +267,19 @@ func (h *Handler) BlockTransactionCount(id BlockID) (uint64, *jsonrpc.Error) {
 	return header.TransactionCount, nil
 }
 
+// GetNodesFromRoot is the new RPC method
+func (h *Handler) GetNodesFromRoot( key *felt.Felt) ([]trie.StorageNode, *jsonrpc.Error) {
+    convertedKey := h.trie.FeltToKey(key) // Call the feltToKey function
+    nodes, err := h.trie.NodesFromRoot(&convertedKey) // Pass a pointer to NodesFromRoot
+    if err != nil {
+        return nil, &jsonrpc.Error{
+            Code:    jsonrpc.InternalError,
+            Message: err.Error(),
+        }
+    }
+	 return nodes, nil
+}
```

Comment:
Exercise 5 requires granting the handler access to the global state trie. To do this, try access the headState() using the handlers bcReader. You can then add a method to this interface (eg GlobalStateTrie) to get access to the state trie. Then call GetNodesFromRoot(). You don't need to add a trie field to the handler type, so it can be removed.